src/back/wgsl/writer.rs

```diff
@@ -173,6 +182,30 @@ impl<W: Write> Writer<W> {
         Ok(())
     }
 
+    /// Helper method used to write stuct name
+    ///
+    /// # Notes
+    /// Adds no trailing or leading whitespace
+    fn write_struct_name(&mut self, module: &Module, handle: Handle<Type>) -> BackendResult {
+        if module.types[handle].name.is_none() {
+            #[allow(clippy::pattern_type_mismatch)]
+            let shader_stage = self.ep_results.iter().find(|(_, ty)| *ty == handle);
+            if let Some(stage) = shader_stage {
+                match stage.0 {
```

Comment:
we should move things out of matches where possible, i.e. here it would be `let name = match { .. };` followed up by `write!(self.out, "{}", name)?;`